crates/ruff_linter/src/rules/ruff/rules/unnecessary_cast_to_int.rs

```diff
@@ -0,0 +1,161 @@
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Arguments, Expr, ExprCall, ExprName, ExprNumberLiteral, Number};
+use ruff_python_semantic::analyze::typing;
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::TextRange;
+
+/// ## What it does
+/// Checks for `int` conversions of values that are already integers.
+///
+/// ## Why is this bad?
+/// Such a conversion is unnecessary.
+///
+/// ## Known problems
+/// When values incorrectly override the `__round__`, `__ceil__`, `__floor__`,
```

Comment:
Such overrides are not necessarily incorrect. [The documentation](https://docs.python.org/3/reference/datamodel.html#object.__round__) says “all these methods should return the value of the object truncated to an [`Integral`](https://docs.python.org/3/library/numbers.html#numbers.Integral) (typically an [`int`](https://docs.python.org/3/library/functions.html#int))”. They don’t need to return instances of `int` specifically, so calling `int` on the result might sometimes be necessary.

Reply:
Good point. I'm sorry, that I misguided you. In that case my suggestion is to:

* Change the fix back to unsafe for the cases where this applies
* Add a fix safety section and explain for which method calls the fix is unsafe.

Reply:
Yeah, agreed. Sorry from me too!